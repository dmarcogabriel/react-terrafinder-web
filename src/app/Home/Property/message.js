export const message = (ownerName, propertyName) => {
  return `Ol√°, ${ownerName}! Estou interessado na sua propriedade chamada ${propertyName}!
  Qual a melhor forma de pagamento?
  `;
};
