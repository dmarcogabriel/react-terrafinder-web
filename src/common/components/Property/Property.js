import React from 'react';
import { moneyFormat, farmingFormat } from 'utils/formatters';
import imagePlaceholder from 'common/static/soja.jpg';
import { Card, CardMedia, Box, Typography } from '@mui/material';
import {
  FullscreenOutlined as SizeIcon,
  Map as MapIcon,
  Spa as FarmingIcon,
  Star as StarIcon,
  Edit as EditIcon,
} from '@mui/icons-material';
import {
  PropertyAmount,
  PropertyAtributesBox,
  PropertyAttribute,
  PropertyPremiumLabel,
  PropertyEditButton,
} from './styles';

export const Property = ({
  dataTestId,
  property,
  onSelect,
  isPremium,
  isEditable,
  onClickEdit,
  containerSx = {},
}) => {
  const [photo] = property.photos;

  const handleSelect = () => onSelect && onSelect(property._id);

  const handleEdit = () => onClickEdit && onClickEdit(property._id);

  return (
    <Card
      data-testid={dataTestId || 'property'}
      sx={{ my: 1, cursor: 'pointer', position: 'relative', ...containerSx }}
    >
      {isPremium && (
        <PropertyPremiumLabel sx={{ p: 1 }}>
          <StarIcon fontSize="small" sx={{ mr: 1 }} />
          An√∫ncio Premium
        </PropertyPremiumLabel>
      )}
      {isEditable && (
        <PropertyEditButton
          variant="contained"
          color="secondary"
          onClick={handleEdit}
        >
          <EditIcon fontSize="large" />
        </PropertyEditButton>
      )}
      <CardMedia
        image={
          photo
            ? `${process.env.REACT_APP_STATIC}/images/${photo}`
            : imagePlaceholder
        }
        alt={property.name}
        sx={{ height: 200 }}
      />
      <Box
        sx={{
          py: 2,
          pl: 2,
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'space-between',
        }}
        onClick={handleSelect}
      >
        <Typography variant="h6">{property.name}</Typography>
        <PropertyAmount sx={{ p: 1 }}>
          {moneyFormat(property.amount, false)}
        </PropertyAmount>
      </Box>
      <PropertyAtributesBox sx={{ py: 2, px: 2 }}>
        <PropertyAttribute sx={{ my: { xs: 1, md: 0 } }}>
          <SizeIcon color="success" />
          <Typography sx={{ ml: 1 }}>{property.size}ha</Typography>
        </PropertyAttribute>
        <PropertyAttribute sx={{ my: { xs: 1, md: 0 } }}>
          <FarmingIcon color="success" />
          <Typography sx={{ ml: 1 }}>
            {farmingFormat(property.farming)}
          </Typography>
        </PropertyAttribute>
        <PropertyAttribute sx={{ my: { xs: 1, md: 0 } }}>
          <MapIcon color="success" />
          <Typography sx={{ ml: 1 }}>{property.state}</Typography>
        </PropertyAttribute>
      </PropertyAtributesBox>
    </Card>
  );
};
