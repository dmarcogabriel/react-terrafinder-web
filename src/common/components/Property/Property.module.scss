@import 'styles';

.propertyItem {
  box-shadow: $shadow-base;
  margin: 1.25rem 0;
  cursor: pointer;

  > img {
    width: 100%;
  }

  .content {
    position: relative;
    padding: 1rem;

    .name {
      font-size: 1.25rem;
      margin: 0.5rem 0;
    }

    .amount {
      display: inline-block;
      padding: 0.5rem;
      background: $primary-dark;
      color: $white;
      border-radius: 0.2rem;
      margin: 0.5rem 0;
      font-size: 1.25rem;

      @media ($desktop) {
        position: absolute;
        top: 1rem;
        right: 0;
        border-bottom-right-radius: 0;
        border-top-right-radius: 0;
      }
    }

    .attributes {
      @media ($desktop) {
        display: flex;
        justify-content: space-between;
        width: 70%;
      }

      .attribute {
        display: flex;
        align-items: center;
        margin: 0.5rem 0;

        > p {
          margin-left: 1rem;
          font-weight: bold;
        }

        .icon {
          color: $green;
        }
      }
    }
  }
}
