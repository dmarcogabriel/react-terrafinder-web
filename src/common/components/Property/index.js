import React from 'react';
import { moneyFormat, farmingFormat } from 'utils/formatters';
import imagePlaceholder from 'common/static/soja.jpg';
import { Card, CardMedia, Box, Typography } from '@mui/material';
import {
  FullscreenOutlined as SizeIcon,
  Map as MapIcon,
  Spa as FarmingIcon,
  Star as StarIcon,
} from '@mui/icons-material';
import {
  PropertyAmount,
  PropertyAtributesBox,
  PropertyAttribute,
  PropertyPremiumLabel,
} from './styles';

export default function Property({
  dataTestId,
  property,
  onSelect,
  isPremium,
  containerSx = {},
}) {
  const [photo] = property.photos;

  const handleSelect = () => onSelect && onSelect(property._id);

  return (
    <Card
      data-testid={dataTestId || 'property'}
      onClick={handleSelect}
      sx={{ my: 1, cursor: 'pointer', position: 'relative', ...containerSx }}
    >
      {isPremium && (
        <PropertyPremiumLabel sx={{ p: 1 }}>
          <StarIcon fontSize="small" sx={{ mr: 1 }} />
          An√∫ncio Premium
        </PropertyPremiumLabel>
      )}
      <CardMedia
        image={
          photo
            ? `${process.env.REACT_APP_STATIC}/images/${photo}`
            : imagePlaceholder
        }
        alt={property.name}
        sx={{ height: 200 }}
      />
      <Box
        sx={{
          py: 2,
          pl: 2,
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'space-between',
        }}
      >
        <Typography variant="h6">{property.name}</Typography>
        <PropertyAmount sx={{ p: 1 }}>
          {moneyFormat(property.amount, false)}
        </PropertyAmount>
      </Box>
      <PropertyAtributesBox sx={{ py: 2, px: 2 }}>
        <PropertyAttribute sx={{ my: { xs: 1, md: 0 } }}>
          <SizeIcon color="success" />
          <Typography sx={{ ml: 1 }}>{property.size}ha</Typography>
        </PropertyAttribute>
        <PropertyAttribute sx={{ my: { xs: 1, md: 0 } }}>
          <FarmingIcon color="success" />
          <Typography sx={{ ml: 1 }}>
            {farmingFormat(property.farming)}
          </Typography>
        </PropertyAttribute>
        <PropertyAttribute sx={{ my: { xs: 1, md: 0 } }}>
          <MapIcon color="success" />
          <Typography sx={{ ml: 1 }}>{property.state}</Typography>
        </PropertyAttribute>
      </PropertyAtributesBox>
    </Card>
  );
}
