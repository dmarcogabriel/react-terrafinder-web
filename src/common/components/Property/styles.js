import React from 'react';
import { styled } from '@mui/material/styles';
import { Typography, Box, Button, ButtonBase } from '@mui/material';

export const ClickableArea = styled(ButtonBase)({
  // styles here...
});

export const PropertyPremiumLabel = styled(Typography)({
  backgroundColor: '#ffcb08',
  borderBottomLeftRadius: '0.2rem',
  borderBottomRightRadius: '0.2rem',
  position: 'absolute',
  left: '1.5rem',
  display: 'flex',
  alignItems: 'center',
});

export const PropertyAmount = styled(({ children, ...props }) => (
  <Typography {...props}>{children}</Typography>
))(({ theme }) => ({
  backgroundColor: '#ffcb08',
  borderTopLeftRadius: '0.2rem',
  borderBottomLeftRadius: '0.2rem',

  [theme.breakpoints.up('md')]: {
    // position: absolute;
    // top: 1rem;
    // right: 0;
    // border-bottom-right-radius: 0;
    // border-top-right-radius: 0;
  },
}));

export const PropertyAtributesBox = styled(Box)(({ theme }) => ({
  [theme.breakpoints.up('md')]: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
}));

export const PropertyAttribute = styled(Box)({
  display: 'flex',
  alignItems: 'center',
});
