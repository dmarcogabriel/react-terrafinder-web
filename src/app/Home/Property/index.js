import React, { useEffect, useState } from 'react';
import { Button, Input } from 'common/components';
import { MdArrowForward } from 'react-icons/md';
import { useRouteMatch } from 'react-router-dom';
import api from 'services/api';
import TextArea from 'common/components/TextArea';
import classes from './Property.module.scss';
import propertyImg from './static/background-banner1.jpg';
import ownerPlaceholderImg from './static/ownerPlaceholder.png';

export default function Property() {
  const { params } = useRouteMatch();
  const [property, setProperty] = useState();
  const [owner, setOwner] = useState();
  const [loading, setLoading] = useState(true);

  const loadProperty = async () => {
    setLoading(true);
    const res = await api.get(`property/${params.id}`);

    setProperty(res.data.property);
    setOwner(res.data.property.user);
    setLoading(false);
  };

  useEffect(() => {
    loadProperty();
  }, []);

  return (
    !loading && (
      <div className={classes.container}>
        <h1>{property.name}</h1>

        <div className={classes.content}>
          <div className={classes.propertySession}>
            <div className={classes.propertyImage}>
              <img src={propertyImg} alt="" />
            </div>

            <div className={classes.col}>
              <div className={classes.row}>
                <div className={classes.session}>
                  <p className={classes.title}>Resumo</p>

                  <p>Valor: R$ 11.000,00</p>
                  <p>Área: 300ha</p>
                  <p>Código: {property._id}</p>
                  <p>Cidade mais próxima: {property.nearbyCity}</p>
                  <p>Estado: Mato Grosso do Sul</p>
                  <p>Tipo de Propriedade: {property.propertyKind}</p>
                  <p>Cultivo: Soja / Milho</p>
                </div>

                <div className={classes.session}>
                  <p className={classes.title}>Atividades e Destaques</p>

                  <ul>
                    <li>Casa com 2 quartos</li>
                    <li>Caseiro</li>
                    <li>Animais domésticos</li>
                    <li>Sinal de rede wifi</li>
                    <li>Temporada de caça legalizada</li>
                    <li>Fácil localização</li>
                    <li>Cultivos alternativos</li>
                    <li>Registrado</li>
                  </ul>
                </div>
              </div>

              <div>
                <div className={classes.session}>
                  <p className={classes.title}>Descrição da Propriedade</p>

                  <p>{property.description}</p>
                </div>
              </div>
            </div>
          </div>

          <div className={classes.ownerSession}>
            <div className={classes.ownerAvatar}>
              <img src={ownerPlaceholderImg} alt="" />
            </div>

            <p
              className={classes.ownerName}
            >{`${owner.firstName} ${owner.lastName}`}</p>

            <p>Entre em contato com o proprietário pelo formulário abaixo:</p>

            <Input label="Nome" placeholder="João da Silva" />
            <Input label="E-mail" placeholder="joao@dasilva.com" />
            <Input label="Telefone" placeholder="(11) 9999-9999" />

            <TextArea label="Escreva aqui sua mensagem para o proprietário" />

            <Button className={classes.sendMessageButton}>
              <p>Enviar mensagem ao proprietário</p>

              <MdArrowForward size={22} />
            </Button>
          </div>
        </div>
      </div>
    )
  );
}
