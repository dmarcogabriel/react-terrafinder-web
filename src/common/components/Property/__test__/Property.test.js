import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import Property from '..';

const property = {
  _id: 'test',
  name: 'Test',
  amount: 99999999,
  size: '20',
  state: 'ZZ',
  photos: ['testphoto.png'],
  farming: ['testFarming'],
};

const mockProperty = {
  property,
  index: 0,
};

const OLD_ENV = process.env;

beforeEach(() => {
  jest.resetModules();
  process.env = { ...OLD_ENV };
});

afterAll(() => {
  process.env = OLD_ENV;
});

describe('<Property />', () => {
  it('renders without crashing', () => {
    process.env.REACT_APP_STATIC = 'http://testinghost:555';

    const div = document.createElement('div');
    ReactDOM.render(<Property {...mockProperty} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render without photo', () => {
    // ! Ta errado isso aqui
    // const { getByTestId } = render(
    //   <Property property={{ ...property, photos: [] }} />
    // );
    // const photo = getByTestId('photo');
    // expect(photo).toHaveAttribute('alt', 'Propriedade');
    // expect(photo).toHaveAttribute('src', JSON.stringify('soja.jpg'));
  });

  it('should fire onselect event', () => {
    const mockFn = jest.fn();
    const { getByTestId } = render(
      <Property property={{ ...property, photos: [] }} onSelect={mockFn} />
    );

    const propertyDiv = getByTestId('property');
    fireEvent.click(propertyDiv);

    expect(mockFn).toHaveBeenCalled();
    expect(mockFn.mock.calls).toHaveLength(1);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Property {...mockProperty} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
