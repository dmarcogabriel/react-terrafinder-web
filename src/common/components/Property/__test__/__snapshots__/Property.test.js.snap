// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Property /> matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-b24pkv-MuiPaper-root-MuiCard-root"
  data-testid="property"
  onClick={[Function]}
>
  <div
    alt="Test"
    className="MuiCardMedia-root css-muxve-MuiCardMedia-root"
    role="img"
    style={
      Object {
        "backgroundImage": "url(\\"undefined/images/testphoto.png\\")",
      }
    }
  />
  <div
    className="MuiBox-root css-imun1j"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Test
    </h6>
    <p
      className="MuiTypography-root MuiTypography-body1 css-afhmsr-MuiTypography-root"
    >
      R$Â 99.999.999
    </p>
  </div>
  <div
    className="MuiBox-root css-nb7qyz"
  >
    <div
      className="MuiBox-root css-16h10fm"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-4b67si-MuiSvgIcon-root"
        data-testid="FullscreenOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
        />
      </svg>
      <p
        className="MuiTypography-root MuiTypography-body1 css-yyvcc2-MuiTypography-root"
      >
        20
        ha
      </p>
    </div>
    <div
      className="MuiBox-root css-16h10fm"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-4b67si-MuiSvgIcon-root"
        data-testid="SpaIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.55 12c-1.07-.71-2.25-1.27-3.53-1.61 1.28.34 2.46.9 3.53 1.61zm10.43-1.61c-1.29.34-2.49.91-3.57 1.64 1.08-.73 2.28-1.3 3.57-1.64z"
        />
        <path
          d="M15.49 9.63c-.18-2.79-1.31-5.51-3.43-7.63-2.14 2.14-3.32 4.86-3.55 7.63 1.28.68 2.46 1.56 3.49 2.63 1.03-1.06 2.21-1.94 3.49-2.63zm-6.5 2.65c-.14-.1-.3-.19-.45-.29.15.11.31.19.45.29zm6.42-.25c-.13.09-.27.16-.4.26.13-.1.27-.17.4-.26zM12 15.45C9.85 12.17 6.18 10 2 10c0 5.32 3.36 9.82 8.03 11.49.63.23 1.29.4 1.97.51.68-.12 1.33-.29 1.97-.51C18.64 19.82 22 15.32 22 10c-4.18 0-7.85 2.17-10 5.45z"
        />
      </svg>
      <p
        className="MuiTypography-root MuiTypography-body1 css-yyvcc2-MuiTypography-root"
      >
        testFarming
      </p>
    </div>
    <div
      className="MuiBox-root css-16h10fm"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-4b67si-MuiSvgIcon-root"
        data-testid="MapIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m20.5 3-.16.03L15 5.1 9 3 3.36 4.9c-.21.07-.36.25-.36.48V20.5c0 .28.22.5.5.5l.16-.03L9 18.9l6 2.1 5.64-1.9c.21-.07.36-.25.36-.48V3.5c0-.28-.22-.5-.5-.5zM15 19l-6-2.11V5l6 2.11V19z"
        />
      </svg>
      <p
        className="MuiTypography-root MuiTypography-body1 css-yyvcc2-MuiTypography-root"
      >
        ZZ
      </p>
    </div>
  </div>
</div>
`;
